LucidTests/Utils/AsyncTaskQueueTests.swift

```diff
@@ -0,0 +1,316 @@
+//
+//  AsyncTaskQueueTests.swift
+//  LucidTests
+//
+//  Created by Stephane Magne on 2023-05-12.
+//  Copyright © 2023 Scribd. All rights reserved.
+//
+
+import XCTest
+@testable import Lucid
+
+final class AsyncTaskQueueTests: XCTestCase {
+
+    // MARK: Adding to empty queue
+
+    func test_that_it_immediately_runs_a_concurrent_operation_added_to_an_empty_queue() {
+
+        let asyncTaskQueue = AsyncTaskQueue(maxConcurrentTasks: 1)
+        let operation = BlockingOperation()
+
+        Task {
+            do {
+                try await asyncTaskQueue.enqueue(operation: {
+                    await operation.setUp()
+                    await operation.perform()
+                })
+            } catch {
+                XCTFail("unexpected error thrown: \(error)")
+            }
+        }
+
+        let setUpExpectation = expectation(description: "set_up_expectation")
```

Comment:
If you make the test func async, you can remove the expectation here. That's part of thinking the `async` way haha

Reply:
See previous comment. If there's a better way to write this as an `async` function, I'd be interested to learn that. Perhaps I'm stuck in the old way of thinking for unit tests.